components/ui/__tests__/carousel.test.tsx

```
@@ -0,0 +1,763 @@
+    });
+  });
+
+  describe('Error Handling', () => {
+    it('should throw error when useCarousel is used outside Carousel context', () => {
+      // Mock console.error to avoid test output pollution
+      const consoleSpy = vi.spyOn(console, 'error').mockImplementation(() => {});
+
+      const TestComponent = () => {
+        try {
+          // This should be imported from the carousel file, but for testing we'll simulate it
+          const context = React.useContext(React.createContext(null));
```

Biome (2.1.2)

[error] This hook is being called conditionally, but all hooks must be called in the exact same order in every component render.

For React to preserve state between calls, hooks needs to be called unconditionally and always in the same order.
See https://reactjs.org/docs/hooks-rules.html#only-call-hooks-at-the-top-level

(lint/correctness/useHookAtTopLevel)